File: 1909-buildings-with-an-ocean-view/buildings-with-an-ocean-view.js
// Stack
// TIme: O(n)
// Space: O(1)...if we don't count the output array
var findBuildings = function(heights) {
    let stack = []
    for (let i = 0; i < heights.length; i++) {
        while (stack.length && heights[stack[stack.length-1]] <= heights[i]) {
            stack.pop();
        }
        stack.push(i)
    }
    return stack;
};

File: 1909-buildings-with-an-ocean-view/buildings-with-an-ocean-view.swift
class Solution {
    func findBuildings(_ heights: [Int]) -> [Int] {
        var stack:[Int] = []

        for (i, n) in heights.enumerated() {
            let curr: Int = heights[i]

            while !stack.isEmpty && heights[stack[stack.count - 1]] <= curr {
                stack.popLast()
            }
            stack.append(i)
        }
        return stack
    }
}